inc/saigroup.h

```diff
@@ -0,0 +1,826 @@
+        _In_ sai_object_id_t ipmc_group_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get IPMC Group attribute
+ *
+ * @param[in] sai_object_id_t IPMC group id
+ * @param[in] attr_count -Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_get_ipmc_group_attribute_fn)(
+        _In_ sai_object_id_t ipmc_group_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Create IPMC group member
+ *
+ * @param[out] ipmc_group_member_id - IPMC group member id
+ * @param[in] attr_count - number of attributes
+ * @param[in] attr_list - array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_create_ipmc_group_member_fn)(
+    _Out_ sai_object_id_t* ipmc_group_member_id,
+    _In_ uint32_t attr_count,
+    _In_ const sai_attribute_t *attr_list
+    );
+
+/**
+ * @brief Remove IPMC group member
+ *
+ * @param[in] ipmc_group_member_id - IPMC group member id
+ *
+ * @return SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_remove_ipmc_group_member_fn)(
+    _In_ sai_object_id_t ipmc_group_member_id
+    );
+
+/**
+ * @brief Set IPMC Group attribute
+ *
+ * @param[in] sai_object_id_t - IPMC group member id
+ * @param[in] attr - attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_set_ipmc_group_member_attribute_fn)(
+    _In_ sai_object_id_t ipmc_group_member_id,
+    _In_ const sai_attribute_t *attr
+    );
+
+/**
+ * @brief Get IPMC Group attribute
+ *
+ * @param[in] sai_object_id_t - ipmc_group_member_id
+ * @param[in] attr_count - number of attributes
+ * @param[inout] attr_list - array of attributes
+ *
+ * @return SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_get_ipmc_group_member_attribute_fn)(
+    _In_ sai_object_id_t ipmc_group_member_id,
+    _In_ uint32_t attr_count,
+    _Inout_ sai_attribute_t *attr_list
+    );
+
+/**
+ *  @brief Next Hop methods table retrieved with sai_api_query()
+ */
+typedef struct _sai_group_api_t
+{
+    sai_create_next_hop_group_fn               create_next_hop_group;
+    sai_remove_next_hop_group_fn               remove_next_hop_group;
+    sai_set_next_hop_group_attribute_fn        set_next_hop_group_attribute;
+    sai_get_next_hop_group_attribute_fn        get_next_hop_group_attribute;
+    sai_create_next_hop_group_member_fn        create_next_hop_group_member;
+    sai_remove_next_hop_group_member_fn        remove_next_hop_group_member;
+    sai_set_next_hop_group_member_attribute_fn set_next_hop_group_member_attribute;
+    sai_get_next_hop_group_member_attribute_fn get_next_hop_group_member_attribute;
+
+    sai_create_rpf_group_fn                    create_rpf_group;
```

Comment:
Other than the same file, we dont seem to reuse any of the existing group things. If we need to add all these new APIs why not create a new file and objects like RPF group **mc etc as part of that.